base/tools/src/main/java/com/netscape/cmstools/HttpClient.java

```diff
@@ -151,96 +165,85 @@ public void send(String ifilename, String ofilename, String tokenName, String db
-                    }
-                    if (sum == 5) {
-                        startSaving = true;
-                        continue;
+    private void sendRequest(String servlet, DataOutputStream dos, String ifilename) throws IOException {
+        byte[] b = getBytesFromFile(ifilename);
+        System.out.println("Total number of bytes read = " + b.length);
+        // send request
+        if (servlet == null) {
+            System.out.println("Missing servlet name.");
+            printUsage();
+        } else {
+            System.out.println("writing to socket");
+            String s = "POST " + servlet + " HTTP/1.0\r\n";
+            dos.writeBytes(s);
+        }
+        dos.writeBytes("Content-length: " + b.length + "\r\n");
+        dos.writeBytes("\r\n");
+        dos.write(b);
+        dos.flush();
+    }
+
+    private void handleResponse(String ofilename, InputStream is, int numHeaderLines) {
+
+        boolean startSaving = false;
+        int sum = 0;
+        boolean hack = false;
+        String catchHeaders = "";
+        try (FileOutputStream fof = new FileOutputStream(ofilename);
+                ByteArrayOutputStream bs = new ByteArrayOutputStream();
+                PrintStream ps = new PrintStream(bs)) {
+            while (true) {
+                int r = is.read();
+                if (r == -1)
+                    break;
+                if (r == 10) { // line break
+                    sum++;
+                }
+
+                // catch the header lines for debugging purposes
+                if (sum < numHeaderLines) {
+                    catchHeaders = catchHeaders + (char) r;
+                }
+                if (sum == 5) {
+                    startSaving = true;
+                    continue;
+                }
+                if (startSaving) {
+                    if (hack) {
+                        fof.write(r);
                     }
-                    if (startSaving) {
-                        if (hack) {
-                            fof.write(r);
-                        }
-                        if (hack == false) {
-                            hack = true;
-                        }
+                    if (hack == false) {
+                        hack = true;
                     }
                 }
-            } catch (IOException e) {
             }
             // debug
             System.out.println("\n##Response Headers begin##\n" + catchHeaders + "\n##end##\n");
 
             byte[] bout = getBytesFromFile(ofilename);
             System.out.println("Total number of bytes read = " + bout.length);
 
-            bs = new ByteArrayOutputStream();
-            ps = new PrintStream(bs);
             ps.print(Utils.base64encode(bout, true));
+
             System.out.println(bs.toString());
 
             System.out.println("");
             System.out.println("The response in binary format is stored in " + ofilename);
             System.out.println("");
```

Comment:
In the original code any exception that happens in this part of the code will be caught and rethrown. In the new code the exception will get swallowed. Should we just rethrow the exception?

Reply:
Good spot, I dropped the empty `catch` and let the calling code deal with the exception.